File: 2606_바이러스.cpp
#include <iostream>
#include <vector>

using namespace std;

int cnt, n, c1, c2, result = 0;
vector<int> list[101];
bool visited[101];

// n번 노드에대하여 dfs
void dfs(int n) {
    // 1. 이미 방문한 노드
    if (visited[n] == true) return;
    // 2. 방문하지 않은 노드
    else {
        // 일단 방문 체크
        visited[n] = true;
        // 방문됬으니까 결과값 +1
        result += 1;
        // 방문된 노드를 기준으로 자식이 있으면 연결된 놈들도 싹 다 판별
        for (int i = 0; i < list[n].size(); i++) {
            dfs(list[n][i]);
        }
    }
}

int main() {
    ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    cin >> cnt;
    cin >> n;
    
    for (int i = 0; i < n; i++) {
        cin >> c1 >> c2;
        list[c1].push_back(c2);
        list[c2].push_back(c1);
      // 얘때문에 틀렸습니다
//         visited[i] = false;
    }
    
    dfs(1);
    
    cout << result - 1 << endl;
}

File: 2606_바이러스.swift
// 알고리즘 스터디 풀이

import Foundation

let num = Int(readLine()!)!
let lineCnt = Int(readLine()!)!
var graph = [[Int]](repeating: [Int](), count: num+1)
var visit = [Bool](repeating: false, count: num+1)
var result = 0

for _ in 0..<lineCnt {
    let node = readLine()!.split(separator: " ").map { Int($0)! }
    graph[node[0]].append(node[1])
    graph[node[1]].append(node[0])
}

dfs(1)
print(result - 1)

func dfs(_ n: Int) {
    if visit[n] { return }
    visit[n] = true
    result += 1
    
    for node in graph[n] {
        dfs(node)
    }
}

// import Foundation

// let n = Int(readLine()!)!
// let m = Int(readLine()!)!
// var list = [[Int]](repeating: [Int](), count: n+1)
// var visited = [Bool](repeating: false, count: n+1)
// var result = 0

// func dfs(n: Int) {
//     if !visited[n] {
//         visited[n] = true
//         result += 1
//         for i in 0..<list[n].count {
//             dfs(n: list[n][i])
//         }
//     }
// }

// for _ in 0..<m {
//     let nodes = readLine()!.components(separatedBy: " ").map{Int(String($0))!}
//     list[nodes[0]].append(nodes[1])
//     list[nodes[1]].append(nodes[0])
// }

// dfs(n: 1)

// print(result - 1)
